ml-agents/mlagents/trainers/curriculum.py

```diff
@@ -44,11 +42,6 @@ def __init__(self, location, default_reset_parameters):
 
         parameters = self.data["parameters"]
         for key in parameters:
-            if key not in default_reset_parameters:
```

Comment:
Do we have a corresponding check (or at least warning) for this now?

Reply:
(at reset time, not init time)

Reply:
Unfortunately, the call on curriculum to get a configuration `get_config(self, lesson=None)` does not have the environment, or the reset_parameters as input. The curriculum does not have a way to check. I would put the check in FloatPropertiesSideChannel, but then I do not have a way to know if reset has been called on the environment yet.

Reply:
OK. Doesn't have to be in this PR, but I think we should consider the Academy declaring the properties it knows about, and complain (either on python or C#) if it gets a property it doesn't recognize.